hnn_core/dipole.py

```diff
@@ -198,15 +249,67 @@ def convert_fAm_to_nAm(self):
     def scale(self, fctr):
         for key in self.data.keys():
             self.data[key] *= fctr
-        return fctr
+        return self
 
-    def smooth(self, winsz):
-        # XXX: add check to make sure self.times is
-        # not smaller than winsz
-        if winsz <= 1:
-            return
-        for key in self.data.keys():
-            self.data[key] = _hammfilt(self.data[key], winsz)
+    def smooth(self, *, window_len=None, h_freq=None):
+        """Smooth the dipole waveform using one of two methods
+        
+        Pass the window length-argument to convolve the data with a Hamming
+        window of the desired length. Alternatively, pass the high frequency
+        argument to apply a Savitzky-Golay filter, which will remove frequency
+        components above this cutoff value (see `~scipy.signal.savgol_filter).
+
+        Note that this method operates in-place, i.e., it will alter the data.
+        If you prefer a filtered copy, consider using the
+        `~hnn_core.dipole.copy`-method.
+
+        Parameters
+        ----------
+        window_len : float or None
+            The length (in ms) of a `~numpy.hamming` window to convolve the
+            data with.
+        h_freq : float or None
+            Approximate high cutoff frequency in Hz. Note that this
+            is not an exact cutoff, since Savitzky-Golay filtering
+            is done using polynomial fits
+            instead of FIR/IIR filtering. This parameter is thus used to
+            determine the length of the window over which a 5th-order
+            polynomial smoothing is applied.
+
+        Returns
+        -------
+        dpl_copy : instance of Dipole
+            A copy of the modified Dipole instance.        
+        """
+        if window_len is None and h_freq is None:
+            raise ValueError('either window_len or h_freq must be defined')
+        elif window_len is not None and h_freq is not None:
+            raise ValueError('set window_len or h_freq, not both')
+            
+        if window_len is not None:
+            winsz = 1e-3 * window_len * self.sfreq
+            if winsz > len(self.times):
+                raise ValueError(
+                    f'Window length too long: {winsz} samples; data length is '
+                    f'{len(self.times)} samples')
+            elif winsz <= 1:
+                # XXX this is to allow param-files with len==0
```

Comment:
When does this situation arise?

Reply:
When applying `postproc=True` with a `params`-file that sets the smoothing to zero... I _really_ want to get rid of that default behaviour!